source/main.cpp

```diff
@@ -45,6 +45,7 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 #include <thread>
 
 #include <cassert>
+#include <future>
```

Comment:
Was this statement for debugging (triggering the issue) and should it be removed?

Reply:
This was a missing include from #6582.